ca/server.go

```diff
@@ -464,7 +467,12 @@ func (s *Server) Run(ctx context.Context) error {
 // Stop stops the CA and closes all grpc streams.
 func (s *Server) Stop() error {
 	s.mu.Lock()
+
+	// Wait for Run to complete before returning
+	defer s.wg.Wait()
```

Comment:
It may not matter, since I don't think we're checking the error returned by `Stop`, but would this mean that `Stop()` always returns an error?